tests/tests_unit/test_base.py

```diff
@@ -239,6 +240,42 @@ def test_to_pandas_no_camels(self):
         expected_df = pd.DataFrame({"var_a": [1, 2], "var_b": [None, 3]})
         pd.testing.assert_frame_equal(resource_list.to_pandas(camel_case=False), expected_df)
 
+    @pytest.mark.dsl
+    def test_to_pandas_metadata(self):
+        import pandas as pd
+
+        event_list = EventList(
+            [
+                Event(external_id="ev1", metadata={"value1": 1, "value2": "hello"}),
+                Event(external_id="ev2", metadata={"value1": 2, "value2": "world"}),
+            ]
+        )
+
+        expected_df = pd.DataFrame(
+            data={"external_id": ["ev1", "ev2"], "metadata.value1": [1, 2], "metadata.value2": ["hello", "world"]},
+        )
+
+        actual_df = event_list.to_pandas(expand_metadata=True)
+        pd.testing.assert_frame_equal(expected_df, actual_df, check_like=False)
+
+    @pytest.mark.dsl
+    def test_to_pandas_metadata_some_nulls(self):
+        import pandas as pd
+
+        event_list = EventList(
+            [Event(external_id="ev1", metadata={"val1": 1}), Event(external_id="ev2", metadata={"val2": 2})]
+        )
+        expected_df = pd.DataFrame(
+            data={
+                "external_id": ["ev1", "ev2"],
+                "metadata.val1": [1, None],
+                "metadata.val2": [None, 2],
+            }
+        )
```

Comment:
I assume this is what you meant with some nulls @haakonvt?

Reply:
Looks good!